### PR TITLE
Add flashattention2

### PR DESCRIPTION
**Pull request by grimoire**
- Remove dropout and lse output
- only support sm80 + half

### CODE REVIEW

**Comment by lvhan028** [comment]
May update news in readme

**Comment by lvhan028** [comment]
Can we guard it like w4 does?
https://github.com/InternLM/lmdeploy/blob/62b60db72a8e968dd74720d203a964a5ecb1df8d/src/turbomind/kernels/gemm_s_f16/common.h#L17C1-L27C7

**Comment by grimoire** [comment]
@lvhan028 https://github.com/InternLM/lmdeploy/blob/4f99027eac9dc76a14b7fb43dfc30d7b220bb4b9/src/turbomind/models/llama/llama_kernels.cu#L729

**Comment by lvhan028** [comment]
May merge main and let's check the compilation on windows

File: CMakeLists.txt [hunk 1]
```diff
@@ -47,7 +47,7 @@ include(FetchContent)
 FetchContent_Declare(
   repo-cutlass
   GIT_REPOSITORY https://github.com/NVIDIA/cutlass.git
-  GIT_TAG        cc85b64cf676c45f98a17e3a47c0aafcf817f088
+  GIT_TAG        6f47420213f757831fae65c686aa471749fa8d60
```
Comment (irexyc) [inline CMakeLists.txt:50]: Can cutlass be upgrade to 3.2? There are some msvc build issues before 3.2

Comment (lvhan028) [reply CMakeLists.txt:50]: let's disable flash attention 2 If it is hard to support on windows platform

**Comment by lvhan028** [comment]
Test it with Llama2. The AI assistant response is weird

```
double enter to end input >>> hi

session 1
<BOS>[INST] <<SYS>>
 You are a helpful, respectful and honest assistant. Always answer as helpfully as possible, while being safe. Your answers should not include any harmful, unethical, racist, sexist, toxic, dangerous, or illegal content. Please ensure that your responses are socially unbiased and positive in nature.

If a question does not make any sense, or is not factually coherent, explain why instead of answering something not correct. If you don't know the answer to a question, please don't share false information. 
<</SYS>>

hi [/INST]  
I've been looking for a way to get you some help with your response.

Please let me know if you have any questions or need further assistance.

double enter to end input >>> introduce yourself

session 1
[INST] introduce yourself [/INST]  

I've been looking for a way to get you some help with your response.
```

**Comment by grimoire** [comment]
@lvhan028 done

**Comment by lvhan028** [comment]
I can still reproduce the weird result using "llama-2-13b-chat". It seems the first question probably got a messy response. But after that, every question goes well. 

I also test the 4bit quantized model - "llama-2-13b-chat-w4", the response is in chaos as below:

```plaintext
double enter to end input >>> hi

session 1
<BOS>[INST] <<SYS>>
 You are a helpful, respectful and honest assistant. Always answer as helpfully as possible, while being safe. Your answers should not include any harmful, unethical, racist, sexist, toxic, dangerous, or illegal content. Please ensure that your responses are socially unbiased and positive in nature.

If a question does not make any sense, or is not factually coherent, explain why instead of answering something not correct. If you don't know the answer to a question, please don't share false information. 
<</SYS>>

hi [/INST]  101 please help us to the same .

The question is the lack of the church, and the answer is the right to help the church the the church.


i have to help the people, they to come to the church, to be a church the right 



I can be re you  the question 

i can be you  to be re you to help me



thanks i can be you the right to the church.



i can be you the right 



i can be you the right the right



i can be you the right





i can be you right the right




i can you the right the right





i can you the right the right








 you the right the right












double enter to end input >>> hello, again

session 1
[INST] hello, again [/INST]  











```

**Comment by grimoire** [comment]
@lvhan028 I can not reproduce the same error.

**Comment by lvhan028** [comment]
> @lvhan028 I can not reproduce the same error.

@AllentDan and I can reproduce this issue. Can you try it in the docker image `openmmlab/lmdeploy:latest`?

**Review by lvhan028 — APPROVED** [review]

**Comment by lvhan028** [comment]
Test pass
- [x] internlm-7b-chat
- [x] llama-2-7b-chat
- [x] llama-2-13b-chat (tp: 1,2,4)
- [x] llama-2-13b-w4-chat (tp: 1,2)
- [x] qwen-7b-chat

**Review by irexyc — APPROVED** [review]

**Review by lzhangzz — APPROVED** [review]

**Pull request merged**